user-lock-manager-w-keypad-support.smartapp.groovy

```diff
@@ -1066,20 +1092,29 @@ def performActions(evt) {
     def codeData = new JsonSlurper().parseText(evt.data)
     if(enabledUsersArray().contains(codeData.usedCode) || isManualUnlock(codeData)) {
       // Global Hello Home
-      if (noRunPresence && doRunPresence == null) {
-        if (!anyoneHome(noRunPresence)) {
-          location.helloHome.execute(homePhrases)
-        }
-      } else if (doRunPresence && noRunPresence == null) {
-        if (anyoneHome(doRunPresence)) {
-          location.helloHome.execute(homePhrases)
-        }
-      } else if (doRunPresence && noRunPresence) {
-        if (anyoneHome(doRunPresence) && !anyoneHome(noRunPresence)) {
-          location.helloHome.execute(homePhrases)
-        }
-      } else {
-       location.helloHome.execute(homePhrases)
+      if(location.currentMode != homePhrases)
```

Comment:
Currently this check isn't working properly. The idea is to prevent the SmartApp from running a Routine if it's already in the mode that it's trying to change to. This is an issue when you have multiple users without presence devices and they keep adjusting the thermostat every time someone unlocks a door.